src/components/MemberList.tsx

```diff
@@ -1,39 +1,217 @@
+import { Github, Linkedin, Code2 } from 'lucide-react';
 import { Member } from '../types';
 import { resolveName } from './utils/RandomUtils';
 
 interface MemberListProps {
   members: Member[];
+  loading?: boolean;
+  currentPage?: number;
+  totalPages?: number;
+  showPagination?: boolean;
+  onPageChange?: (page: number) => void;
 }
 
-const MemberList: React.FC<MemberListProps> = ({ members }) => {
+const MemberCard = ({ member }: { member: Member }) => {
+  const bgColor = useColorModeValue('white', 'gray.800');
+  const borderColor = useColorModeValue('gray.200', 'gray.700');
+  const iconColor = useColorModeValue('gray.600', 'gray.400');
+
+  const fullName = resolveName(member);
+
+  return (
+    <Box
+      p={6}
+      borderWidth={1}
+      borderRadius="lg"
+      borderColor={borderColor}
+      bg={bgColor}
+      boxShadow="sm"
+      _hover={{ boxShadow: 'md' }}
+      transition="box-shadow 0.2s"
+      minH="140px"
+      w="100%"
+    >
+      <HStack spacing={4} align="start" h="100%">
+        <Avatar size="lg" name={fullName} src={member.profilePictureUrl} />
+        <Box flex="1">
+          <Flex justify="space-between" align="start">
+            <Box>
+              <Text fontWeight="bold" fontSize="lg">
+                {fullName}
+              </Text>
+              <Text color="gray.500" fontSize="sm">
+                @{member.username}
+              </Text>
+              {member.major && (
+                <Text color="gray.600" fontSize="sm" mt={1}>
+                  {member.major}
+                </Text>
+              )}
+            </Box>
+
+            <HStack spacing={2}>
+              {member.github?.username && !member.github?.isPrivate && (
+                <Tooltip label="GitHub Profile">
+                  <IconButton
+                    as="a"
+                    href={`https://github.com/${member.github.username}`}
+                    target="_blank"
+                    rel="noopener noreferrer"
+                    aria-label="GitHub Profile"
+                    icon={<Github size={20} />}
+                    variant="ghost"
+                    color={iconColor}
+                    size="sm"
+                  />
+                </Tooltip>
+              )}
+
+              {member.linkedin?.username && !member.linkedin?.isPrivate && (
+                <Tooltip label="LinkedIn Profile">
+                  <IconButton
+                    as="a"
+                    href={`https://linkedin.com/in/${member.linkedin.username}`}
+                    target="_blank"
+                    rel="noopener noreferrer"
+                    aria-label="LinkedIn Profile"
+                    icon={<Linkedin size={20} />}
+                    variant="ghost"
+                    color={iconColor}
+                    size="sm"
+                  />
+                </Tooltip>
+              )}
+
+              {member.leetcode?.username && !member.leetcode?.isPrivate && (
```

Comment:
Are we transmitting the leetcode profiles etc if they are private? Someone can just open console and view the network call if not and it's not private anymore. If it's not transmitted then this second check is not needed.

If it can be viewed in the network call I'd say we shouldn't even have the option then because that's a false sense of privacy.